packages/dart_mappable_builder/lib/src/builders/mappable_builder.dart

```diff
@@ -138,6 +137,7 @@ class MappableBuilder implements Builder {
 
     output.write('}');
 
+    final outputId = buildStep.allowedOutputs.last;
```

Comment:
Please also adjust the code regarding `writeImports` (see line 125 and 153) , since this currently assumes the output is under the same path as the input.